### PR TITLE
Ghetto Seed Extraction with tools

### PR DESCRIPTION
**Pull request by Ms-Mee**
## About The Pull Request

This Pull Request implements a way to extract seeds from produce with tools, time and sound effects may need changing, however the current state is usable and does not seem to have issues.

## Why It's Good For The Game

Given the sheer emphasis on ghetto analogues for processes and how boring placing produce in a seed extractor to get seeds is, I see this as a nice addition that makes ghetto botany slightly more accessible.

## Changelog
:cl:
add: Added a means of extracting seeds using sharp objects and tools. As well as a chance for composting produce to plant the same type of produce on an empty tray.
balance: New ghetto options for seed extraction and planting unreliant on autolathes.
/:cl:

### CODE REVIEW

**Comment by triplezeta** [comment]
yes

File: code/modules/hydroponics/grown.dm [hunk 1]
```diff
@@ -85,6 +85,18 @@
 			for(var/datum/plant_gene/trait/T in seed.genes)
 				T.on_attackby(src, O, user)
 
+//Ghetto Seed Extraction
+/obj/item/reagent_containers/food/snacks/grown/attackby(obj/item/W, mob/user, params)
+	if(W.get_sharpness())
+		playsound(loc, 'sound/weapons/slice.ogg', 50, TRUE, -1)
+		user.visible_message("<span class='notice'>[user] starts slicing apart \the [src].</span>", "<span class='notice'>You start slicing apart \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object slicing some plant matter.</span>")
+		if(do_after(user, 50, target = src))
+			to_chat(user, "<span class='notice'>You slice apart this [src].</span>")
```
Comment (mcmeiler) [inline code/modules/hydroponics/grown.dm]:
```suggestion
			to_chat(user, "<span class='notice'>You slice apart [src].</span>")
```
Pretty sure ``src`` already contains a "the"
grafik

Comment (Ms-Mee) [reply code/modules/hydroponics/grown.dm]: Yeah I'll change it once I have the other tools in place and all, not quite as it's intended just there so I don't forget what's what

File: code/modules/hydroponics/grown.dm [hunk 2]
```diff
@@ -85,6 +85,49 @@
...
+		if(W.tool_behaviour == TOOL_WRENCH)
+			playsound(loc, 'sound/weapons/smash.ogg', 50, TRUE, -1)
+			user.visible_message("<span class='notice'>[user] starts wackng \the [src].</span>", "<span class='notice'>You start wacking \the [src]...</span>", "<span class='hear'>You hear the sound of a plant being wacked violently.</span>")
+			if(do_after(user, 50, target = src))
+				to_chat(user, "<span class='notice'>You smash [src]! Sadly there's nothing left of it other than the seeds.</span>")
+			/*	new /obj/item/reagent_containers/food/snacks/grown/smacked(user.drop_location(), 1)
+					use(1) */
+				seedify(src, 1)
+		else
+			return ..()
```
Comment (mcmeiler) [inline code/modules/hydroponics/grown.dm:130]:
```suggestion
/obj/item/reagent_containers/food/snacks/grown/attackby(obj/item/W, mob/user, params)
	if(W.get_sharpness())
		playsound(loc, 'sound/weapons/slice.ogg', 50, TRUE, -1)
		user.visible_message("<span class='notice'>[user] starts slicing apart \the [src].</span>", "<span class='notice'>You start slicing apart \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object slicing some plant matter.</span>")
		if(do_after(user, 50, target = src))
			to_chat(user, "<span class='notice'>You slice apart the [src]! You went too far and the tiny remaining scraps are worthless!</span>")
		/*	new /obj/item/reagent_containers/food/snacks/grown/sliced(user.drop_location(), 1)
			use(1) */
			seedify(src, 1)
	switch(W.tool_behaviour)
		if(TOOL_SCREWDRIVER)
			playsound(loc, 'sound/weapons/bite.ogg', 50, TRUE, -1)
			user.visible_message("<span class='notice'>[user] starts digging into \the [src].</span>", "<span class='notice'>You start digging into \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object penetrating some plant matter.</span>")
			if(do_after(user, 50, target = src))
				to_chat(user, "<span class='notice'>You dig into the [src] to collect it's seeds! It's all gross and unusuable now, ew!</span>")
			/*	new /obj/item/reagent_containers/food/snacks/grown/chipped(user.drop_location(), 1)
				use(1) */
				seedify(src, 1)
		if(TOOL_WIRECUTTER)
			playsound(loc, 'sound/weapons/bite.ogg', 50, TRUE, -1)
			user.visible_message("<span class='notice'>[user] starts chipping into \the [src].</span>", "<span class='notice'>You start chipping into \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object penetrating some plant matter.</span>")
			if(do_after(user, 50, target = src))
				to_chat(user, "<span class='notice'>You dig into the [src] to collect it's seeds! It's all gross and unusuable now, ew!</span>")
				seedify(src, 1)
		if(TOOL_CROWBAR)
			playsound(loc, 'sound/weapons/slice.ogg', 50, TRUE, -1)
			user.visible_message("<span class='notice'>[user] starts splitting \the [src].</span>", "<span class='notice'>You dig into \the [src] and start to split it...</span>", "<span class='hear'>You hear the sound of a sharp object digging into some plant matter.</span>")
			if(do_after(user, 50, target = src))
				to_chat(user, "<span class='notice'>You split apart the [src]! Sadly you put too much force and it's remains are unusable, but hey, you got your seeds!</span>")
			/*	new /obj/item/reagent_containers/food/snacks/grown/split(user.drop_location(), 1)
				use(1) */
				seedify(src, 1)
		if(TOOL_WRENCH)
			playsound(loc, 'sound/weapons/smash.ogg', 50, TRUE, -1)
			user.visible_message("<span class='notice'>[user] starts wackng \the [src].</span>", "<span class='notice'>You start wacking \the [src]...</span>", "<span class='hear'>You hear the sound of a plant being wacked violently.</span>")
			if(do_after(user, 50, target = src))
				to_chat(user, "<span class='notice'>You smash [src]! Sadly there's nothing left of it other than the seeds.</span>")
			/*	new /obj/item/reagent_containers/food/snacks/grown/smacked(user.drop_location(), 1)
					use(1) */
				seedify(src, 1)
		else
			return ..()
```

File: code/modules/hydroponics/grown.dm [hunk 3]
```diff
@@ -85,6 +85,49 @@
...
+				seedify(src, 1)
+		if(W.tool_behaviour == TOOL_SCREWDRIVER)
+			playsound(loc, 'sound/weapons/bite.ogg', 50, TRUE, -1)
+			user.visible_message("<span class='notice'>[user] starts digging into \the [src].</span>", "<span class='notice'>You start digging into \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object penetrating some plant matter.</span>")
+			if(do_after(user, 50, target = src))
+				to_chat(user, "<span class='notice'>You dig into the [src] to collect it's seeds! It's all gross and unusuable now, ew!</span>")
+			/*	new /obj/item/reagent_containers/food/snacks/grown/chipped(user.drop_location(), 1)
+				use(1) */
+				seedify(src, 1)
+		if(W.tool_behaviour == TOOL_WIRECUTTER)
```
Comment (mcmeiler) [inline code/modules/hydroponics/grown.dm]:
```suggestion
		if(TOOL_WIRECUTTER)

File: code/modules/hydroponics/grown.dm [hunk 4]
```diff
@@ -85,6 +85,49 @@
...
+			playsound(loc, 'sound/weapons/bite.ogg', 50, TRUE, -1)
+			user.visible_message("<span class='notice'>[user] starts chipping into \the [src].</span>", "<span class='notice'>You start chipping into \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object penetrating some plant matter.</span>")
+			if(do_after(user, 50, target = src))
+				to_chat(user, "<span class='notice'>You dig into the [src] to collect it's seeds! It's all gross and unusuable now, ew!</span>")
+				seedify(src, 1)
+		if(W.tool_behaviour == TOOL_CROWBAR)
```
Comment (mcmeiler) [inline code/modules/hydroponics/grown.dm]:
```suggestion
		if(TOOL_CROWBAR)

File: code/modules/hydroponics/grown.dm [hunk 5]
```diff
@@ -85,6 +85,49 @@
...
+			playsound(loc, 'sound/weapons/slice.ogg', 50, TRUE, -1)
+			user.visible_message("<span class='notice'>[user] starts splitting \the [src].</span>", "<span class='notice'>You dig into \the [src] and start to split it...</span>", "<span class='hear'>You hear the sound of a sharp object digging into some plant matter.</span>")
+			if(do_after(user, 50, target = src))
+				to_chat(user, "<span class='notice'>You split apart the [src]! Sadly you put too much force and it's remains are unusable, but hey, you got your seeds!</span>")
+			/*	new /obj/item/reagent_containers/food/snacks/grown/split(user.drop_location(), 1)
+				use(1) */
+				seedify(src, 1)
+		if(W.tool_behaviour == TOOL_WRENCH)
```
Comment (mcmeiler) [inline code/modules/hydroponics/grown.dm]:
```suggestion
		if(TOOL_WRENCH)

Comment (mcmeiler) [reply code/modules/hydroponics/grown.dm]: Keep in mind what that you may have to do manual fixing if you commit these changes.

These changes fixed compiling for me in VSC, and they still work in-game.

File: code/modules/hydroponics/grown.dm (hunk 3 above)
Comment (mcmeiler) [reply code/modules/hydroponics/grown.dm]: Unnecessary for ``switch()``

File: code/modules/hydroponics/grown.dm [hunk 6]
```diff
@@ -85,6 +85,49 @@
...
+/obj/item/reagent_containers/food/snacks/grown/attackby(obj/item/W, mob/user, params)
+	switch(W)
+		if(W.get_sharpness())
+			playsound(loc, 'sound/weapons/slice.ogg', 50, TRUE, -1)
+			user.visible_message("<span class='notice'>[user] starts slicing apart \the [src].</span>", "<span class='notice'>You start slicing apart \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object slicing some plant matter.</span>")
+			if(do_after(user, 50, target = src))
+				to_chat(user, "<span class='notice'>You slice apart the [src]! You went too far and the tiny remaining scraps are worthless!</span>")
+			/*	new /obj/item/reagent_containers/food/snacks/grown/sliced(user.drop_location(), 1)
+				use(1) */
+				seedify(src, 1)
```
Comment (mcmeiler) [inline code/modules/hydroponics/grown.dm]:
```suggestion
	if(W.get_sharpness())
		playsound(loc, 'sound/weapons/slice.ogg', 50, TRUE, -1)
		user.visible_message("<span class='notice'>[user] starts slicing apart \the [src].</span>", "<span class='notice'>You start slicing apart \the [src]...</span>", "<span class='hear'>You hear the sound of a sharp object slicing some plant matter.</span>")
		if(do_after(user, 50, target = src))
			to_chat(user, "<span class='notice'>You slice apart the [src]! You went too far and the tiny remaining scraps are worthless!</span>")
		/*	new /obj/item/reagent_containers/food/snacks/grown/sliced(user.drop_location(), 1)
			use(1) */
			seedify(src, 1)
	switch(W.tool_behaviour)

Comment (mcmeiler) [reply code/modules/hydroponics/grown.dm]: Removing the faulty ``switch()`` and putting the right one at the bottom, also fixing indentation.

File: code/modules/hydroponics/grown.dm [hunk 7]
```diff
@@ -85,6 +85,49 @@
...
+		if(W.tool_behaviour == TOOL_SCREWDRIVER)
```
Comment (mcmeiler) [inline code/modules/hydroponics/grown.dm]:
```suggestion
		if(TOOL_SCREWDRIVER)

**Review by mcmeiler — CHANGES_REQUESTED** [review]

**Comment by Ms-Mee** [comment]
A deal has been struck with a particularly twisted easter deity, now all that is left is the leftover items and adjusting the time and sounds, which is optional.
Unless checks fails, I hope such a fate comes not

File: code/modules/hydroponics/grown.dm [hunk 8]
```diff
@@ -85,6 +85,71 @@
 			for(var/datum/plant_gene/trait/T in seed.genes)
 				T.on_attackby(src, O, user)
 
+/proc/flamingmoai(obj/item/O, t_max, mob/living/user, var/d_item)
```
Comment (MarkSuckerberg) [inline code/modules/hydroponics/grown.dm]: 🗿

**Review by MarkSuckerberg — APPROVED** [review]

**Comment by Ms-Mee** [comment]
Sure hope this does not break, it should not, if it does, I WILL scream. You have been warned.

**Pull request merged**